src/core/providers/strategies/LedgerProviderStrategy/LedgerProviderStrategy.ts

```diff
@@ -86,24 +83,46 @@ export class LedgerProviderStrategy {
     const provider = this.provider as unknown as IProvider;
     provider.setAccount({ address: this.address });
     provider.signTransactions = this.signTransactions;
-    provider.login = this.login;
     provider.signMessage = this.signMessage;
+    provider.login = this.login;
+    provider.cancelLogin = this.cancelLogin;
 
     return provider;
   };
 
-  private initialize = () => {
-    if (this.address) {
-      return;
+  private ledgerLogin = async (
+    options?: LoginOptionsTypes
+  ): Promise<{ address: string; signature: string }> => {
+    if (!this.provider) {
+      throw new Error(ProviderErrorsEnum.notInitialized);
     }
+    if (!options || typeof options.addressIndex !== 'number') {
+      throw new Error('Missing addressIndex for Ledger login');
+    }
+    const { address, signature } = await this.provider.login({
+      addressIndex: options.addressIndex
+    });
+    return {
+      address,
+      signature: signature || ''
+    };
+  };
 
-    const address = getAddress();
-
-    if (!address) {
-      return;
+  public override getLoginOperation = async (options?: LoginOptionsTypes) => {
+    if (!this.provider) {
+      throw new Error(ProviderErrorsEnum.notInitialized);
     }
+    const ledgerConnectManager = LedgerConnectStateManager.getInstance();
+    await ledgerConnectManager.init();
 
-    this.address = address;
+    return await authenticateLedgerAccount({
```

Comment:
actually the result of await is not an operation, it's a value
getLoginOperation sounds like it's returning a function. And see that it has no return type adnotation